cmd/kubeadm/app/preflight/checks.go

```diff
@@ -394,6 +394,27 @@ func (sysver SystemVerificationCheck) Check() (warnings, errors []error) {
 	return warns, nil
 }
 
+// SwapCheck warns if swap is enabled
+type SwapCheck struct{}
+
+func (swc SwapCheck) Check() (warnings, errors []error) {
+	f, err := os.Open("/proc/swaps")
+	if err != nil {
+		// /procs/swaps not available, thus no reasons to warn
+		return nil, nil
+	}
+	defer f.Close()
+	var buf []string
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		buf = append(buf, scanner.Text())
+	}
+	if len(buf) > 1 {
+		return []error{fmt.Errorf("Running with swap on is not supported. Please disable swap or set kubelet's --fail-swap-on flag to false.")}, nil
```

Comment:
Note the return values order. it will return that message as warning now (errors is nil).
Kubelet actually will fail by default, but if user use `--fail-swap-on=false` it will work, so this check is inteded to be warning, as I don't see reason to force user to turn off swap if user knows consequences.